FirebaseRemoteConfig/Sources/RCNFetch.m

```diff
@@ -215,14 +215,27 @@ - (void)refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:
                     @"Failed to register InstanceID with error : %@.", error);
       }
       if (token) {
-        NSError *IIDError;
         strongSelf->_settings.configInstanceIDToken = [token copy];
-        strongSelf->_settings.configInstanceID = [instanceID appInstanceID:&IIDError];
-        FIRLogInfo(kFIRLoggerRemoteConfig, @"I-RCN000022", @"Success to get iid : %@.",
-                   strongSelf->_settings.configInstanceID);
+        __weak RCNConfigFetch *instanceIDSelf = strongSelf;
+        [instanceID getIDWithHandler:^(NSString *_Nullable identity, NSError *_Nullable error) {
+          RCNConfigFetch *instanceIDStrongSelf = instanceIDSelf;
+          instanceIDStrongSelf->_settings.configInstanceID = identity;
+
+          if (identity && !error) {
+            FIRLogInfo(kFIRLoggerRemoteConfig, @"I-RCN000022", @"Success to get iid : %@.",
+                       instanceIDStrongSelf->_settings.configInstanceID);
+          } else {
+            FIRLogWarning(kFIRLoggerRemoteConfig, @"I-RCN000055", @"Error getting iid : %@.",
```

Comment:
Is it possible for !error && !identity? If so, this will crash.

Reply:
yes it is..not sure where we would crash though..